Userland/Libraries/LibJS/Runtime/Value.cpp

```diff
@@ -474,6 +474,63 @@ FLATTEN ThrowCompletionOr<Value> Value::to_numeric(GlobalObject& global_object)
     return primitive.to_number(global_object);
 }
 
+struct NumberParseResult {
+    StringView literal;
+    u8 base;
+    bool invalid { false };
+};
+
+static Optional<NumberParseResult> parse_number_text(StringView text)
+{
+    NumberParseResult result {};
+
+    auto check_prefix = [&](auto lower_prefix, auto upper_prefix) {
```

Comment:
In general I would like to see some reference to the spec here.
Now it is unclear whether we should even do this base conversion in ToNumber.
For this function you could for example link https://tc39.es/ecma262/#sec-tonumber-applied-to-the-string-type